core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt

```diff
@@ -1,12 +1,13 @@
 package com.strumenta.kolasu.model
 
+import java.io.Serializable
 import kotlin.reflect.KClass
 import kotlin.reflect.KProperty1
 import kotlin.reflect.KVisibility
 import kotlin.reflect.full.findAnnotation
 import kotlin.reflect.full.memberProperties
 
-interface Origin {
+interface Origin : Serializable {
```

Comment:
Is this necessary? Not all Origins will be serializable...

Reply:
Yes, but a Node could specify an Origin or a Destination, so all Origins and all Destinations should be Serializable

Reply:
ParseTreeOrigin isn't, though, for example. IMHO it's more appropriate if only actual serializable concrete classes implement Serializable.

Reply:
You are right, I make only specific subclasses Serializable now